protocol/contracts/farm/facets/SiloFacet/SiloFacet.sol

```diff
@@ -82,25 +82,66 @@ contract SiloFacet is TokenSilo {
      */
 
     function transferDeposit(
+        address sender,
         address recipient,
         address token,
         uint32 season,
         uint256 amount
-    ) external payable nonReentrant updateSilo {
+    ) external payable nonReentrant returns (uint256 bdv) {
+        if (sender != msg.sender) {
+            _spendDepositAllowance(sender, msg.sender, token, amount);
+        }
+        _update(sender);
         // Need to update the recipient's Silo as well.
         _update(recipient);
-        _transferDeposit(msg.sender, recipient, token, season, amount);
+        bdv = _transferDeposit(sender, recipient, token, season, amount);
     }
 
     function transferDeposits(
+        address sender,
         address recipient,
         address token,
         uint32[] calldata seasons,
         uint256[] calldata amounts
-    ) external payable nonReentrant updateSilo {
+    ) external payable nonReentrant returns (uint256[] memory bdvs) {
+        require(amounts.length > 0, "Silo: amounts array is empty");
+        for (uint256 i = 0; i < amounts.length; i++) {
+            require(amounts[i] > 0, "Silo: amount in array is 0");
+            if (sender != msg.sender) {
```

Comment:
Gas optimization: move this check outside the loop, since `sender` and `msg.sender` don't change across iterations.